Orchestrator with BedrockAugmentedLLM is failing

Hi, I am trying the orchestrator example with BedrockAugmentedLLM and it failed initially with the following error

`[ERROR] 2025-06-03T17:03:02 mcp_agent.workflows.llm.augmented_llm_bedrock.LLM Orchestration Planner - Error: An error occurred (ValidationException) when calling the Converse operation: 
The maximum tokens you requested exceeds the model limit of 10000. Try again with a maximum tokens value that is lower than 10000.`

After this, I tried setting the maxTokens param in the RequestParams like the following
`result = await orchestrator.generate_str(message=task, request_params=RequestParams(model="amazon.nova-pro-v1:0", maxTokens=10000))`

Now I started getting the following error

```
Traceback (most recent call last):
  File "/Users/ks/work/projects/mcp_github_test/orchestrator_sample.py", line 173, in <module>
    asyncio.run(example_usage())
  File "/opt/homebrew/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/ks/work/projects/mcp_github_test/orchestrator_sample.py", line 163, in example_usage
    result = await orchestrator.generate_str(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mcp_agent/workflows/orchestrator/orchestrator.py", line 131, in generate_str
    result = await self.generate(
             ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mcp_agent/workflows/orchestrator/orchestrator.py", line 120, in generate
    plan_result = await self.execute(objective=objective, request_params=params)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mcp_agent/workflows/orchestrator/orchestrator.py", line 186, in execute
    plan = await self._get_full_plan(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mcp_agent/workflows/orchestrator/orchestrator.py", line 331, in _get_full_plan
    plan = await self.planner.generate_structured(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mcp_agent/workflows/llm/augmented_llm_bedrock.py", line 304, in generate_structured
    structured_response = client.chat.completions.create(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/instructor/client.py", line 366, in create
    return self.create_fn(
           ^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/instructor/patch.py", line 187, in new_create_sync
    response_model, new_kwargs = handle_response_model(
                                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/instructor/process_response.py", line 1127, in handle_response_model
    response_model, new_kwargs = mode_handlers[mode](response_model, new_kwargs)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/instructor/process_response.py", line 839, in handle_bedrock_json
    new_kwargs = _prepare_bedrock_converse_kwargs_internal(new_kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/instructor/process_response.py", line 820, in _prepare_bedrock_converse_kwargs_internal
    raise NotImplementedError(
NotImplementedError: Non-text prompts are not currently supported in the Bedrock provider.
```

The response received from the Bedrock is

```
{
  "data": {
    "ResponseMetadata": {
      "RequestId": "02a03029-xxxx-478e-ad53-7b134ad20814",
      "HTTPStatusCode": 200,
      "HTTPHeaders": {
        "date": "Tue, 03 Jun 2025 11:34:03 GMT",
        "content-type": "application/json",
        "content-length": "2404",
        "connection": "keep-alive",
        "x-amzn-requestid": "02a03029-xxxx-478e-ad53-7b134ad20814"
      },
      "RetryAttempts": 0
    },
    "output": {
      "message": {
        "role": "assistant",
        "content": [
          {
            "text": "{\n    \"steps\": [\n        {\n            \"description\": \"Load the student's short story and relevant style guides.\",\n            \"tasks\": [\n                
{\n                    \"description\": \"Load the student's short story from short_story.md.\",\n                    \"agent\": \"finder\"\n                },\n                {\n        
\"description\": \"Fetch the APA style guide on formatting from https://apastyle.apa.org/learn/quick-guide-on-formatting.\",\n                    \"agent\": \"finder\"\n                
},\n                {\n                    \"description\": \"Fetch the APA style guide on references from https://apastyle.apa.org/learn/quick-guide-on-references.\",\n                   
\"agent\": \"finder\"\n                }\n            ]\n        },\n        {\n            \"description\": \"Analyze the short story for proofreading, factuality/logical consistency, and
style adherence.\",\n            \"tasks\": [\n                {\n                    \"description\": \"Proofread the short story for grammar, spelling, and punctuation errors.\",\n      
\"agent\": \"proofreader\"\n                },\n                {\n                    \"description\": \"Check the factual consistency within the story.\",\n                    \"agent\":
\"fact_checker\"\n                },\n                {\n                    \"description\": \"Analyze the story for adherence to APA style guidelines.\",\n                    \"agent\": 
\"style_enforcer\"\n                }\n            ]\n        },\n        {\n            \"description\": \"Generate a report with feedback and write it to graded_report.md.\",\n          
\"tasks\": [\n                {\n                    \"description\": \"Compile the feedback from proofreading, factuality/logical consistency, and style adherence into a report.\",\n     
\"agent\": \"writer\"\n                },\n                {\n                    \"description\": \"Write the graded report to graded_report.md in the same directory as 
short_story.md.\",\n                    \"agent\": \"writer\"\n                }\n            ]\n        }\n    ],\n    \"is_complete\": false\n}"
          }
        ]
      }
    },
    "stopReason": "end_turn",
    "usage": {
      "inputTokens": 1611,
      "outputTokens": 390,
      "totalTokens": 2001
    },
    "metrics": {
      "latencyMs": 5338
    }
  }
}
```
I am guessing this is probably due to the `content` being already a list type and where as the instructor dependency has this piece of code where it throws an error if the content is not of string type. Any workaround that can be made here? Or shall I raise the issue with the instructor repo?

mcp-agent version: 0.0.23
instructor version: 1.8.3

--- a/src/mcp_agent/workflows/llm/augmented_llm_bedrock.py
+++ b/src/mcp_agent/workflows/llm/augmented_llm_bedrock.py
@@ -420,7 +420,7 @@ class BedrockCompletionTasks:
                 aws_access_key_id=request.config.aws_access_key_id,
                 aws_secret_access_key=request.config.aws_secret_access_key,
                 aws_session_token=request.config.aws_session_token,
-                region_name=request.config.bedrock.aws_region,
+                region_name=request.config.aws_region,
             )
         else:
             session = Session()
@@ -431,7 +431,7 @@ class BedrockCompletionTasks:
         # Extract structured data from natural language
         structured_response = client.chat.completions.create(
             modelId=request.model,
-            messages=[{"role": "user", "content": [{"text": request.response_str}]}],
+            messages=[{"role": "user", "content": request.response_str}],
             response_model=response_model,
         )
 

--- a/src/mcp_agent/workflows/llm/augmented_llm_bedrock.py
+++ b/src/mcp_agent/workflows/llm/augmented_llm_bedrock.py
@@ -69,6 +69,13 @@ class BedrockAugmentedLLM(AugmentedLLM[MessageUnionTypeDef, MessageUnionTypeDef]
         if self.context.config.bedrock:
             if hasattr(self.context.config.bedrock, "default_model"):
                 default_model = self.context.config.bedrock.default_model
+        else:
+            self.logger.error(
+                "Bedrock configuration not found. Please provide Bedrock configuration."
+            )
+            raise ValueError(
+                "Bedrock configuration not found. Please provide Bedrock configuration."
+            )
 
         self.default_request_params = self.default_request_params or RequestParams(
             model=default_model,
@@ -377,7 +384,7 @@ class BedrockCompletionTasks:
                 aws_access_key_id=request.config.aws_access_key_id,
                 aws_secret_access_key=request.config.aws_secret_access_key,
                 aws_session_token=request.config.aws_session_token,
-                region_name=request.config.bedrock.aws_region,
+                region_name=request.config.aws_region,
             )
         else:
             session = Session()

